denlillemand/week2/index.html

```diff
@@ -0,0 +1,302 @@
+			if(message.id === messageId) {
+			    messageIndex = index;
+			}
+		    });
+		    if(messageIndex > -1) {
+			let copiedCurrentList = this.state[currentList].slice();
+			let removedMessage = copiedCurrentList.splice(messageIndex, 1);
+			if(removedMessage[0].archived) {
+			    alert("Cannot remove an archived message");
+			} else {
+			    let copiedDestinationList = this.state[destinationList].slice();
+			    this.setState({
+				[currentList]: copiedCurrentList,
+				[destinationList]: copiedDestinationList.concat(removedMessage)
+			    });
+			}
+		    } else {
+			alert("Couldn't find the message id you wanted to remove");
+		    }
+		};
+
+		onSubmitNamedList(name) {
+		    if(this.state[name]) {
+			alert("Cannot create another list with the same name");
+		    } else {
+			this.setState({
+			    [name]: []
+			});
+		    }
+		}
+
+		onSubmitMessage(message, listName) {
+		    let copiedList = this.state[listName].slice();
+		    copiedList.push(message);
+		    this.setState({
+			[listName]: copiedList
+		    });
+		}
+
+		deleteMessage(id, listName) {
+		    let copiedMessages = this.state[listName].slice();
+		    let messageIdToDelete = -1;
+		    copiedMessages.forEach(function(message, index) {
+			if(message.id === id) {
+			    messageIdToDelete = index;
+			}
+		    });
+		    copiedMessages.splice(messageIdToDelete, 1);
+		    this.setState({
+			[listName]: copiedMessages
+		    });
+		}
+
+		render() {
+		    var lists = [];
+		    for(var key in this.state) {
+			lists.push(<li><NamedList onSubmitMessage={this.onSubmitMessage} name={key} messages={this.state[key]} swap={this.swap} archiveMessage={this.archiveMessage} 
+	 deleteMessage={this.deleteMessage} swapMessage={this.swapMessage}/></li>);
+		    }
+		    return(
+			<div>
+			    <h3>Assignment 3</h3>
+			    <div>
+			    	<h4>Create list:</h4>
+				<NamedListForm onSubmitNamedList={this.onSubmitNamedList} />
+			    </div>
+			    <div>
+				<h4>Lists:</h4>
+				<ul>
+				{lists}
+				</ul>
+			    </div>
+			</div>
+		    );
+		}
+	    }
+
+	    class NamedList extends React.Component {
+		constructor(props, context) {
+		    super(props, context);
+		    this.state = {
+			messageSwapDestination: "",
+			message: "",
+			sortedMessages: [],
+			messageSwapId: "",
+			messageId: 1 
+		    };
+		    this.onMessageSubmit = this.onMessageSubmit.bind(this);
+		    this.onChangeMessageSwapDestinationListName = this.onChangeMessageSwapDestinationListName.bind(this);
+		    this.onChangeMessage = this.onChangeMessage.bind(this);
+		    this.onChangeMessageSwapId = this.onChangeMessageSwapId.bind(this);
+		    this.archiveMessage = this.archiveMessage.bind(this);
+		    this.deleteMessage = this.deleteMessage.bind(this);
+		    this.swapMessage = this.swapMessage.bind(this);
+		}
+
+		swapMessage(event) {
+		    this.props.swapMessage(parseInt(this.state.messageSwapId), this.state.messageSwapDestination, this.props.name);
```

Comment:
Event if, when not specified, the `radix` parameter defaults to 10, it is recommended to always specify that parameter. Not specifying `radix` can lead to different behavior due to different implementations of that method.